client/rest-high-level/src/main/java/org/elasticsearch/client/migration/IndexUpgradeRequest.java

```diff
@@ -18,52 +18,47 @@
  */
 package org.elasticsearch.client.migration;
 
-import org.elasticsearch.action.IndicesRequest;
 import org.elasticsearch.action.support.IndicesOptions;
 import org.elasticsearch.client.TimedRequest;
 
-import java.util.Arrays;
 import java.util.Objects;
 
 /**
  * A request for performing Upgrade on Index
  * Part of Migration API
  */
+public class IndexUpgradeRequest extends TimedRequest {
 
-public class IndexUpgradeRequest extends TimedRequest implements IndicesRequest {
-
-    private String[] indices;
-    private IndicesOptions indicesOptions = IndicesOptions.fromOptions(false, true, true, true);
+    private String index;
```

Comment:
Why did we change this from an array of strings to a single one?

Reply:
for other kind of requests you can specify indices comma separated like _xpack/migration/assistance/index1,index2,index3
but for upgrade only one is allowed (as per documentation)